### PR TITLE
Fix some typos

### PR DESCRIPTION
**Pull request by harryzcy**
As title

### CODE REVIEW

**Comment by wxiaoguang** [comment]
`be overridden` ?

**Comment by wolfogre** [comment]
image

Now I'm sure it was fixed by some spell check tools, `PrecendingCharacter` is function defined in external packages.

I'm not a fan of doing this for a big repo all at once. It comes with more risks than benefits.

**Comment by harryzcy** [comment]
Ok, sure. It's checked by spell checker and replaced manually in the editor.

File: contrib/backport/README [hunk 1]
```diff
@@ -11,7 +11,7 @@ The default version will read from `docs/config.yml`. You can override this
 using the option `--version`.
 
 The upstream branches will be fetched, using the remote `origin`. This can
-be overrided using `--upstream`, and fetching can be avoided using
+be overrode using `--upstream`, and fetching can be avoided using
```
Comment (delvh) [inline contrib/backport/README:14]:
```suggestion
be overridden using `--upstream`, and fetching can be avoided using
```

File: docs/content/doc/administration/config-cheat-sheet.en-us.md [hunk 2]
```diff
@@ -1305,7 +1305,7 @@ Defaultly every storage has their default base path like below
 | actions_log       | actions_log/       |
 | actions_artifacts | actions_artifacts/ |
 
-And bucket, basepath or `SERVE_DIRECT` could be special or overrided, if you want to use a different you can:
+And bucket, basepath or `SERVE_DIRECT` could be special or overriden, if you want to use a different you can:
```
Comment (delvh) [inline docs/content/doc/administration/config-cheat-sheet.en-us.md:1308]:
```suggestion
And bucket, basepath or `SERVE_DIRECT` could be special or overridden, if you want to use a different you can:
```

File: modules/packages/rpm/metadata.go [hunk 3]
```diff
@@ -15,7 +15,7 @@ import (
 )
 
 const (
-	PropertyMetadata = "rpm.metdata"
+	PropertyMetadata = "rpm.metadata"
```
Comment (delvh) [inline modules/packages/rpm/metadata.go:18]: Is that change really possible?
I'm not sure if it isn't breaking.

File: templates/swagger/v1_json.tmpl [hunk 4]
```diff
@@ -22008,7 +22008,7 @@
         "author": {
           "$ref": "#/definitions/CommitUser"
         },
-        "commiter": {
+        "committer": {
```
Comment (delvh) [inline templates/swagger/v1_json.tmpl:22011]: Not good.
That means we have a breaking change in this PR.

File: services/pull/merge_merge.go [hunk 5]
```diff
@@ -9,7 +9,7 @@ import (
 	"code.gitea.io/gitea/modules/log"
 )
 
-// doMergeStyleMerge merges the tracking into the current HEAD - which is assumed to tbe staging branch (equal to the pr.BaseBranch)
+// doMergeStyleMerge merges the tracking into the current HEAD - which is assumed to the staging branch (equal to the pr.BaseBranch)
```
Comment (delvh) [inline services/pull/merge_merge.go:12]:
```suggestion
// doMergeStyleMerge merges the tracking into the current HEAD - which is assumed to be the staging branch (equal to the pr.BaseBranch)
```

**Review by delvh — CHANGES_REQUESTED** [review]
Meant to request changes, the API change must be reverted.

**Comment by puni9869** [comment]
Don't we have spell checker in github action which checks all typos?

**Comment by silverwind** [comment]
There is `misspell`, but it doesn't catch much. Something better would be appreciated.

https://github.com/go-gitea/gitea/blob/2f0e79e6393df13930eaa419273d24dc2ef36cfa/Makefile#L311-L313

**Comment by silverwind** [comment]
I think it's worth a try to add https://github.com/facelessuser/pyspelling via poetry and run that in addition to misspell.

**Comment by puni9869** [comment]
> I think it's worth a try to add https://github.com/facelessuser/pyspelling via poetry and run that in addition to misspell.

May be we can consider https://github.com/super-linter/super-linter

**Comment by puni9869** [comment]
> I think it's worth a try to add https://github.com/facelessuser/pyspelling via poetry and run that in addition to misspell.

I can give a try in next PR

**Comment by silverwind** [comment]
> May be we can consider https://github.com/super-linter/super-linter, but its a lil bit heavy, a monolith, we can customize it specfic to our use case.

No, because every task needs to be runnable locally and once you add something specific to GH Actions, you lose the ability to run checks locally which is very important.

**Comment by puni9869** [comment]
> > May be we can consider https://github.com/super-linter/super-linter, but its a lil bit heavy, a monolith, we can customize it specfic to our use case.
> 
> No, because every task needs to be runnable locally and once you add something specific to GH Actions, you lose the ability to run checks locally which is very important.

Yup, thats true. Let's stick to your suggested package. It is also worth of try.

**Comment by yardenshoham** [comment]
I like https://github.com/streetsidesoftware/cspell-cli https://github.com/streetsidesoftware/cspell-action https://github.com/streetsidesoftware/vscode-spell-checker

**Comment by silverwind** [comment]
Ideally it should be something we can install adn depend on via npm, poetry or golang.

**Comment by yardenshoham** [comment]
It's on npm

**Comment by silverwind** [comment]
Ah yes, I was just about to post [cspell](https://github.com/streetsidesoftware/cspell) as well.

**Comment by harryzcy** [comment]
I was using [typos](https://github.com/crate-ci/typos).

**Comment by silverwind** [comment]
I wonder how those two compare. Using rust crate for tools should be in the realm of possibilites at least 😉.

**Pull request closed without merging**